### ISSUE TITLE
Compatibility - [iOS] [safari] - [Fail to scan Barcode in iPhone (safari).]

### ISSUE THREAD

**Issue by minshinehung** [mebjas/html5-qrcode#618]
**Describe the bug**
Fail to scan Barcode in iPhone (safari).
Already set useBarCodeDetectorIfSupported: true

**Describe the browser:**
 - OS: iOS
 - Browser safari
 - Version 15.7.1

**Screenshots**
Please see attachment
![iphone_html5barcode_barcode](https://user-images.githubusercontent.com/118278617/203240082-8b1a98be-48b7-4302-af9b-99458a176a9b.png)

**Comment by mebjas** [comment]
Yes iOS barcode detector seems to be bad at detecting bar codes. If your
primary use case is barcode I recommend setting that flag as false for now.

**Comment by minshinehung** [comment]
Hi minhaz

Thanks for your reply
We setted useBarCodeDetectorIfSupported to false, but it still Failed to scan Barcode in iPhone (safari).

Thanks for your help

Best Regards

min



________________________________
寄件者: minhaz ***@***.***>
寄件日期: 2022年11月22日 下午 03:59
收件者: mebjas/html5-qrcode ***@***.***>
副本: minshinehung ***@***.***>; Author ***@***.***>
主旨: Re: [mebjas/html5-qrcode] Compatibility - [iOS] [safari] - [Fail to scan Barcode in iPhone (safari).] (Issue #618)


Yes iOS barcode detector seems to be bad at detecting bar codes. If your
primary use case is barcode I recommend setting that flag as false for now.

**Comment by mebjas** [comment]
Please share example barcodes.

**Comment by mebjas** [comment]
What type of code is this?

**Comment by mebjas** [comment]
I am able to scan the same CODE_39 data generated here on https://scanapp.org/barcode but not the screenshot you have attached. Looking further.

**Comment by minshinehung** [comment]
Hi minhaz

1.The test site is https://meeting.csie.ntnu.edu.tw/C_html5qrcode_direct_03
2.The scanned barcode is as attachment. Please print it then scan in paper.
3.Our code is as following

<script src="/_other/js/html5-qrcode.min.js"></script>
<script>

  $(document).ready(function(){

      var lastMessage;
      var codeId = 0;
      function onScanSuccess(decodedText, decodedResult) {
      /**
       * If you following the code example of this page by looking at the
       * source code of the demo page - good job!!
       *
       * Tip: update this function with a success callback of your choise.
       */
            if (lastMessage !== decodedText) {
        //alert( decodedText );
            
        Swal.fire({
            html: decodedText,
            allowOutsideClick: false ,
            type: "success"
        }).then(
            function () {

              return;
            }
        );
            }
      }

    var qrboxFunction = function(viewfinderWidth, viewfinderHeight) {
        // Square QR Box, with size = 80% of the min edge.
        var minEdgeSizeThreshold = 250;
        var edgeSizePercentage = 0.75;
        var minEdgeSize = (viewfinderWidth > viewfinderHeight) ?
            viewfinderHeight : viewfinderWidth;
        var qrboxEdgeSize = Math.floor(minEdgeSize * edgeSizePercentage);
        if (qrboxEdgeSize < minEdgeSizeThreshold) {
            if (minEdgeSize < minEdgeSizeThreshold) {
                return {width: minEdgeSize, height: minEdgeSize};
            } else {
                return {
                    width: minEdgeSizeThreshold,
                    height: minEdgeSizeThreshold
                };
            }
        }
        return {width: qrboxEdgeSize, height: qrboxEdgeSize};
    }

      const html5QrcodeScanner = new Html5QrcodeScanner(
          "reader",
          {
              fps: 10,
              qrbox: qrboxFunction,
              // Important notice: this is experimental feature, use it at your
              // own risk. See documentation in
              // mebjas@/html5-qrcode/src/experimental-features.ts
              experimentalFeatures: {
                  useBarCodeDetectorIfSupported: true
              },
              rememberLastUsedCamera: true,
              showTorchButtonIfSupported: true
          });
      html5QrcodeScanner.render(onScanSuccess);

  })
            

</script>


We have been test it in different iPhone service. It is ok for Android to scan barcode.
What should we do?
Thanks for your help

Best Regards

min



________________________________
寄件者: minhaz ***@***.***>
寄件日期: 2022年11月22日 下午 09:02
收件者: mebjas/html5-qrcode ***@***.***>
副本: minshinehung ***@***.***>; Author ***@***.***>
主旨: Re: [mebjas/html5-qrcode] Compatibility - [iOS] [safari] - [Fail to scan Barcode in iPhone (safari).] (Issue #618)


I am able to scan the same CODE_39 data generated here on https://scanapp.org/barcode but not the screenshot you have attached. Looking further.

**Comment by pawanjenu** [comment]
Any updates on IOS- Barcode scanners?

**Comment by pawanjenu** [comment]
@mebjas , requesting your assistance here. 
thanks

**Comment by jgudo** [comment]
@minshinehung Any updates on this? I'm also facing the same issue with iOS devices. Even the tested it on https://scanapp.org/barcode, still can't scan CODE_39 and CODE_128 barcodes.  However, if I tilt or adjust the camera to a certain angle, it works sometimes, not a good UX though.

**Comment by minshinehung** [comment]
Julius Guevarra:

try this


<div >
      <div class="row">
            <div class="col-12" style="text-align: center;margin-bottom: 20px;">
                  <div id="reader" width="1000" height="1000"></div>
            </div>
      </div>


</div>


<script src="/_other/js/html5-qrcode.min.js"></script>
<script>

  $(document).ready(function(){

    var lastMessage;

    // This method will trigger user permissions
    Html5Qrcode.getCameras().then(devices => {
      if (devices && devices.length) {
        var cameraId = devices[0].id;
        const html5QrCode = new Html5Qrcode("reader");
        const qrCodeSuccessCallback = (decodedText, decodedResult) => {

          if (lastMessage !== decodedText) {
            lastMessage = decodedText;
            alert( decodedText );
          }

        };
        const config = {
          fps: 25,
          qrbox: {
            width: 850,
            height: 300
          },
          experimentalFeatures: {
              useBarCodeDetectorIfSupported: true
          },
          rememberLastUsedCamera: true,
          showTorchButtonIfSupported: true
        };
        html5QrCode.start({
          facingMode: "environment"
        }, config, qrCodeSuccessCallback);
      }
    }).catch(err => {
      // handle err
    });


  })
            

</script>



________________________________
寄件者: Julius Guevarra ***@***.***>
寄件日期: 2023年1月29日 上午 08:15
收件者: mebjas/html5-qrcode ***@***.***>
副本: minshinehung ***@***.***>; Mention ***@***.***>
主旨: Re: [mebjas/html5-qrcode] Compatibility - [iOS] [safari] - [Fail to scan Barcode in iPhone (safari).] (Issue #618)


@minshinehung<https://github.com/minshinehung> Any updates on this? I'm also facing the same issue with iOS devices. Even the tested it on https://scanapp.org/barcode, still can't scan CODE_39 and CODE_128 barcodes. However, if I tilt or adjust the camera to a certain angle, it works sometimes, not a good UX though.

**Comment by GithubOllie** [comment]
I am also having issues that the camera does not scan QR codes when on iPhone/Safari. [Demo](https://imgur.com/MB8EEga)

**Comment by fengpang** [comment]
> I am also having issues that the camera does not scan QR codes when on iPhone/Safari. [Demo](https://imgur.com/MB8EEga)

https://zxing-js.github.io/library/examples/barcode-camera/   
This one can work normally
